open-edge-platform/training_extensions
# Support direct annotation argument (--train/val-ann-files)

### Summary



* Support direct annotation input for CoCo
For now, we can use `--train/val-ann-files` argument
* Support unlabeled file list argument 
* Add unit-tests for both functionalities
* Unifying the `data_root` + `data_roots` --> `data_roots`
* Edit the CHANGELOG
* Edit the documents
![image](https://user-images.githubusercontent.com/80735344/234187929-5a396fbd-37a3-4901-93ca-fbfa530624a4.png)

* Add unlabeled example assets
* Add subset annotation example assets

### How to test
[To check the direct annotation functionality (Dataset Adapter)]
[To check the unlabeled file functionality]
* tox -e pre-merge-all-py310 -- tests/unit/core/data/adapter/test_init.py

[To check the direct annotation functionality (Datumaro function itself)]
* tox -e pre-merge-all-py310 -- tests/unit/core/data/manager/test_dataset_manager.py




### Checklist



- [X] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md)
- [X] I have updated the [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) accordingly

### License

- [X] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```
